schema/query/predicate.go

```diff
@@ -232,7 +232,23 @@ type Equal struct {
 
 // Match implements Expression interface.
 func (e Equal) Match(payload map[string]interface{}) bool {
-	return reflect.DeepEqual(getField(payload, e.Field), e.Value)
+	value := getField(payload, e.Field)
+
+	// The $eq(not present in rest-layer) operator in MongoDB allows matching both single values and an
```

Comment:
I guess you can state "the equality operator" instead of "$eq". What was meant was the implicit equality check as represented by `query.Equal`. Writing `$eq` in the issue was inaccurate.